addons/website_quote/models/sale_order.py

```diff
@@ -1,229 +1,184 @@
+    def write(self, values):
+        values = self._inject_quote_description(values)
+        return super(SaleOrderLine, self).write(values)
 
     # The overrided onchange was written in new api, sorry
     # Take the description on the order template if the product is present in it
     @api.multi
     @api.onchange('product_id')
     def product_id_change(self):
-        domain = super(sale_order_line, self).product_id_change()
+        domain = super(SaleOrderLine, self).product_id_change()
         if self.order_id.template_id:
-            self.name = next((quote_line.name for quote_line in self.order_id.template_id.quote_line if quote_line.product_id.id == self.product_id.id), self.name) 
+            self.name = next((quote_line.name for quote_line in self.order_id.template_id.quote_line if quote_line.product_id.id == self.product_id.id), self.name)
         return domain
 
 
-class sale_order(osv.osv):
+class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    def _get_total(self, cr, uid, ids, name, arg, context=None):
-        res = {}
-        for order in self.browse(cr, uid, ids, context=context):
+    def _default_template_id(self):
+        template = self.env.ref('website_quote.website_quote_template_default', raise_if_not_found=False)
+        return template or False
+
+    access_token = fields.Char('Security Token', required=True, copy=False, default=(lambda self: str(uuid.uuid4())))
+    template_id = fields.Many2one('sale.quote.template', 'Quotation Template', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, default=_default_template_id)
+    website_description = fields.Html('Description', translate=True)
+    options = fields.One2many('sale.order.option', 'order_id', 'Optional Products Lines', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, copy=True)
+    amount_undiscounted = fields.Float(string='Amount Before Discount', compute='_compute_amount_undiscounted', digits=0)
+    quote_viewed = fields.Boolean('Quotation Viewed')
+    require_payment = fields.Selection([
+        (0, 'Not mandatory on website quote validation'),
+        (1, 'Immediate after website order validation'),
+        (2, 'Immediate after website order validation and save a token'),
+    ], 'Payment', help="Require immediate payment by the customer when validating the order from the website quote")
+
+    def _compute_amount_undiscounted(self):
+        for sale_order in self:
             total = 0.0
-            for line in order.order_line:
-                total += line.price_subtotal + line.price_unit * ((line.discount or 0.0) / 100.0) * line.product_uom_qty
-            res[order.id] = total
-        return res
+            for line in sale_order.order_line:
+                total += line.price_subtotal + line.price_unit * ((line.discount or 0.0) / 100.0) * line.product_uom_qty # why is there a discount in a field name amount_undiscounted ??
```

Comment:
Good question ! If it is useless, it should be remove in a REF commit after migration. @tde-banana-odoo isn't weird to include discount in undiscounted amount computation ?